subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/ClasspathFingerprintingStrategy.java

```diff
@@ -223,7 +223,7 @@ public void visit(PhysicalSnapshot fileSnapshot, HashCode normalizedContentHash)
             }
         }
 
-        private FileFingerprint createFileFingerprint(PhysicalSnapshot snapshot, HashCode content) {
+        private FileFingerprint createFileFingerprint(FileSystemLocationSnapshot snapshot, HashCode content) {
```

Comment:
Hmm, I somewhat like `FileFingerprint`... Do you feel strongly about the `FileFingerprint` -> `FileSystemLocationFingerprint` rename?

Reply:
I do, because of the same argument we had about `FileSnapshot`: it's not the fingerprint of a _file,_ but of either a file or a directory. I think it's also desirable that the hierarchies of snapshots and fingerprints line up as much as possible.

Reply:
So now `FileFingerprint` is `FileSystemLocationFingerprint`.